src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -220,6 +220,21 @@ ApplicationSharedPtr FindApp(DataAccessor<ApplicationSet> accessor,
 }
 
 template <class UnaryPredicate>
+ApplicationSharedPtr FindApp(
+    DataAccessor<ApplicationManagerImpl::AppsWaitRegistrationSet> accessor,
+    UnaryPredicate finder) {
+  ApplicationManagerImpl::AppsWaitRegistrationSet::iterator it = std::find_if(
```

Comment:
Unfortunately, if I decide change interface than I should change call this method in all sdl. It take from me to much time.